processor/transaction/event.go

```diff
@@ -34,14 +34,20 @@ func (t *Event) Transform() common.MapStr {
 }
 
 func (t *Event) Mappings(pa *payload) (time.Time, []m.DocMapping) {
-	return t.Timestamp,
-		[]m.DocMapping{
-			{Key: "processor", Apply: func() common.MapStr {
-				return common.MapStr{"name": processorName, "event": t.DocType()}
-			}},
-			{Key: t.DocType(), Apply: t.Transform},
-			{Key: "context", Apply: func() common.MapStr { return t.Context }},
-			{Key: "context.service", Apply: pa.Service.Transform},
-			{Key: "context.system", Apply: pa.System.Transform},
-		}
+	mapping := []m.DocMapping{
+		{Key: "processor", Apply: func() common.MapStr {
+			return common.MapStr{"name": processorName, "event": t.DocType()}
+		}},
+		{Key: t.DocType(), Apply: t.Transform},
+		{Key: "context", Apply: func() common.MapStr { return t.Context }},
+		{Key: "context.service", Apply: pa.Service.Transform},
+		{Key: "context.system", Apply: pa.System.Transform},
+	}
+
+	if pa.Process != nil {
+		mapping = append(mapping, m.DocMapping{
+			Key: "context.process", Apply: pa.Process.Transform,
+		})
+	}
```

Comment:
Why do you handle this differently than for `errors` here https://github.com/elastic/apm-server/pull/445/files#diff-a163837d2241a4bad19d7e4baa4c617bR62?

I suggest to rather add a nil check when processing the mappings, as e.g. `system` is also not a required field.

Reply:
oversight on my part, good catch